frontend/packages/dev-console/src/components/project-access/project-access-form-utils.ts

```diff
@@ -1,24 +1,27 @@
 import * as _ from 'lodash';
-import { UserRoleBinding, UserRole } from './project-access-form-utils-types';
+import { UserRoleBinding, RoleBinding } from './project-access-form-utils-types';
 
-export const filterRoleBindings = (roleBindings, roles) => {
-  return _.filter(roleBindings.data, (user: UserRole) => _.keys(roles).includes(user.roleRef.name));
+export const filterRoleBindings = (roleBindings: RoleBinding[], roles) => {
+  return _.filter(roleBindings, (user: RoleBinding) => _.keys(roles).includes(user.roleRef.name));
 };
 
-export const getUserRoleBindings = (roleBindings) => {
+export const getUsersFromSubject = (user: RoleBinding) => {
+  const users: UserRoleBinding[] = user.subjects?.reduce((acc, obj) => {
+    acc.push({
+      roleBindingName: user.metadata.name,
+      user: obj.name,
+      role: user.roleRef.name,
+    });
+    return acc;
+  }, []);
+
+  return users;
+};
+
+export const getUserRoleBindings = (roleBindings: RoleBinding[]) => {
   let userRoleBindings: UserRoleBinding[] = [];
   roleBindings.map(
-    (user: UserRole) =>
-      (userRoleBindings = [
-        ...userRoleBindings,
-        ...[
-          {
-            roleBindingName: user.metadata.name,
-            user: user.subjects[0].name,
-            role: user.roleRef.name,
-          },
-        ],
-      ]),
+    (user: RoleBinding) => (userRoleBindings = [...userRoleBindings, ...getUsersFromSubject(user)]),
```

Comment:
```suggestion
  return roleBindings.map((user) => getUsersFromSubject(user)).flat();
```
Two major things here:
1. Don't assign in a return statement (we should probably get a linter to stop this)
2. Avoid mutations in the middle of `.map` function calls (the purer they are the better they are)

Reply:
yes, I could have used `flat` here to concatenate the sub-array elements. Thanks!